validator/no_undefined_variables.go

```diff
@@ -0,0 +1,39 @@
+package validator
+
+import "github.com/vektah/gqlparser"
+
+func init() {
+	addRule("NoUndefinedVariables", func(observers *Events, addError addErrFunc) {
+
+		var currentOperation *gqlparser.OperationDefinition
+
+		observers.OnOperation(func(walker *Walker, operation *gqlparser.OperationDefinition) {
+			currentOperation = operation
+		})
+
+		observers.OnOperationLeave(func(walker *Walker, operation *gqlparser.OperationDefinition) {
+			currentOperation = nil
+		})
+
+		observers.OnArgument(func(walker *Walker, arg *gqlparser.Argument) {
```

Comment:
should this just be OnValue?

I guess the thing missing from the test case is something like:
```
 field(foo: {a: $a})
```

Reply:
I just added a commit.
please take a look.
so, I destructured value in rule implementations. Is it right?